build_matrix_A.py

```diff
@@ -0,0 +1,91 @@
+    for m in range(len(mesh)):
+        k[m] = material_property_library[mesh[m][0]]['k']
+        rho[m] = material_property_library[mesh[m][0]]['rho']
+        c[m] = material_property_library[mesh[m][0]]['c']
+        DR[m] = mesh[m][1][1]-mesh[m][1][0]
+
+
+    total_nodes = len(mesh[0][1])
+    for i in range(1, len(mesh)):
+        total_nodes += len(mesh[i][1])-1
+        
+    A = np.zeros((total_nodes, total_nodes))
+    volume = np.zeros(total_nodes)
+    
+    "Begin by setting boundary nodes"
+    "Have to treat nodes 0 and 1 differently to avoid divide by zero"
+    volume[0] = 4/3*np.pi*(mesh[0][1][0]+DR[0]/2)**3
+    volume[1] = 4/3*np.pi*(mesh[0][1][1]+DR[0]/2)**3-(mesh[0][1][1]-DR[0]/2)**3
+    
+    A[0, 0] = k[0]*Dt*4*np.pi/(rho[0]*c[0])/(DR[0]*volume[0])*(DR[0]/2)**2
+    A[0, 1] = -k[0]*Dt*4*np.pi/(rho[0]*c[0])/(DR[0]*volume[0])*(DR[0]/2)**2
+	
+
+    A[1, 1] = k[0]*Dt*4*np.pi/(rho[0]*c[0])/volume[1]*(1/(1/mesh[0][1][1]-1/mesh[0][1][2]) +
+                                    1/DR[0]*(mesh[0][1][0]+DR[0]/2)**2)
+    A[1, 0] = -k[0]*Dt*4*np.pi/(rho[0]*c[0])/(DR[0]*volume[1])*(mesh[0][1][0]+DR[0]/2)**2
+    A[1, 2] = -k[0]*Dt*4*np.pi/(rho[0]*c[0])/volume[1]*(1/(1/mesh[0][1][1]-1/mesh[0][1][2]))
+	
+
+	
+    for i in range(2, len(mesh[0][1])-1): #Length set to fueled node length
+        volume[i] = 4/3*np.pi*((mesh[0][1][i]+DR[0]/2)**3-(mesh[0][1][i]-DR[0]/2)**3)
+        A[i, i] = k[0]*Dt*4*np.pi/(rho[0]*c[0])/volume[i]*(1/(1/mesh[0][1][i]-1/mesh[0][1][i+1]) +
+                                        1/(1/mesh[0][1][i-1]-1/mesh[0][1][i]))
+        A[i, i-1] = -k[0]*Dt*4*np.pi/(rho[0]*c[0])/volume[i]*(1/(1/mesh[0][1][i-1]-1/mesh[0][1][i]))
+        A[i, i+1] = -k[0]*Dt*4*np.pi/(rho[0]*c[0])/volume[i]*(1/(1/mesh[0][1][i]-1/mesh[0][1][i+1]))
+		
+    #Now need to handle the boundary nodes
+    #i is currently the value of the node at the internal boundary
+    for mat in range(1, len(mesh)):
+        prev_mesh_length = len(mesh[mat-1][1])
+        volume[i+1] = 4/3*np.pi*((mesh[mat-1][1][prev_mesh_length-1]+DR[mat]/2)**3-(
+                                                 mesh[mat-1][1][prev_mesh_length-1]-DR[mat-1]/2)**3)
+        A[i+1,i+2] = -k[mat]*Dt*4*np.pi/(rho[mat-1]*c[mat-1])/volume[i+1]*(1/(1/mesh[mat][1][0]-1/mesh[mat][1][1]))
+        A[i+1,i+1] = (k[mat-1]*Dt*4*np.pi/(rho[mat-1]*c[mat-1])/volume[i+1]*(1/(1/mesh[mat-1][1][prev_mesh_length-2]
+                                                    -1/mesh[mat-1][1][prev_mesh_length-1]))+(
+                                                    k[mat]*Dt*4*np.pi/(rho[mat-1]*c[mat-1])/volume[i+1]*(1/(1/mesh[mat][1][0]
+                                                    -1/mesh[mat][1][1]))))
+        A[i+1, i] = -k[mat-1]*Dt*4*np.pi/(rho[mat-1]*c[mat-1])/volume[i+1]*(1/(1/mesh[mat-1][1][prev_mesh_length-2]-
+                                                                        1/mesh[mat-1][1][prev_mesh_length-1]))
+	
+        for j in range(1, len(mesh[mat][1])-1): 
```

Comment:
Maybe add `i` in the definition of the loop to make the below notation easier?

Reply:
There are places where j is used alone (i+j being the location of the node overall, while j is the node number within a given material). That's why I chose to go this way, though I do admit it is a bit messy.

Reply:
perhaps have some other variable for `i+j` or `i+j+1` so that it only needs to be calculated once, avoiding typos, etc